Develop

arkomashina opened the pull request:
Написал задание для 6 спринта

arkomashina committed:
- Написал тесты на классы Lion, Feline, Cat & Alex
- Написал класс Alex
- проверил покрытие с помощью Jacoco

arkomashina committed:
- Написал тесты на классы Lion, Feline, Cat & Alex
- Написал класс Alex
- проверил покрытие с помощью Jacoco

arkomashina committed:
- Доработал тесты
- Добавил обработку Exception в конструкторе класса Lion
- Добавил отчёт Jacoco

alesyaisachenko commented on src/main/java/com/example/Lion.java:
```diff
@@ -5,18 +5,21 @@
     }
 
-    Feline feline = new Feline();
+    public Lion (Feline feline){
```
Это нужно исправить.


изоляцию Lion от Feline нужно было делать через инъекцию зависимостей. То есть в класс Lion должен попадать уже созданный экземпляр класса Feline.
В классе должен быть только  один конструктор public Lion(String sex, Feline feline).

arkomashina replied:
Понял, поправил.

alesyaisachenko commented on src/main/java/com/example/Alex.java:
```diff
@@ -0,0 +1,27 @@
+import java.util.List;
+
+public class Alex extends Lion {
+    Feline feline;
```
Это нужно доработать:

это поле (переменная) не нужно -- оно наследуется из класса Lion

arkomashina replied:
Понял, поправил.

alesyaisachenko commented on src/test/java/FelineTests.java:
```diff
@@ -0,0 +1,48 @@
+    public void getFamilyTest(){
+        feline = new Feline();
+        String actual = feline.getFamily();
+        Assert.assertEquals("Не принадлежит к семейству кошачих", "Кошачьи", actual);
```
Это нужно доработать:

в этом тесте функционал @Spy не используется. Тут или использовать функционал @Spy или использовать другой экземпляр класса

alesyaisachenko commented on src/test/java/FelineTests.java:
```diff
@@ -0,0 +1,48 @@
+    public void ifFelineEatMeatReturnsListOfAnimals() throws Exception {
+        feline = new Feline();
+        List<String> actual = feline.eatMeat();
+        Assert.assertEquals("Этим хищники не питаются", List.of("Животные", "Птицы", "Рыба"), actual);
```
Это нужно доработать:

в этом тесте функционал @Spy не используется. Тут или использовать функционал @Spy или использовать другой экземпляр класса

alesyaisachenko commented on src/test/java/FelineTests.java:
```diff
@@ -0,0 +1,48 @@
+    @Test
+    public void getKittensWithParamsReturnKittensCount(){
+        int actualKittens = feline.getKittens(10);
+        Assert.assertEquals("Кол-во котов не совпадает", 10, actualKittens);
```
Это нужно доработать:

в этом тесте функционал @Spy не используется. Тут или использовать функционал @Spy или использовать другой экземпляр класса

arkomashina replied:
Чуть разобрался, в проверках я его убрал. Но дописал новый тест используя функционал @SPY

alesyaisachenko commented on src/test/java/FelineTests.java:
```diff
@@ -0,0 +1,48 @@
+    @Test
+    public void eatMeatInvokesMethodGetFoodForPredator() throws Exception {
+        feline.eatMeat();
+        Mockito.verify(feline, Mockito.times(1)).getFood("Хищник");
```
Это нужно доработать:
в этом месте не нужно использовать verify , метод getFood() всегда вызывается один раз. Такая проверка не удачная.

alesyaisachenko commented on src/test/java/LionTests.java:
```diff
@@ -0,0 +1,43 @@
+        Assert.assertEquals("Хищники едят не это", actual, List.of("Животные", "Птицы", "Рыба"));
+    }
+
+    @Test
```
Можно лучше:


если в тесте ожидается ассерт, то можно делать так:
@Test(expected = Exception.class)


в этом случае блок try-catch не нужен

arkomashina committed:
- Поправил класс Lion, использовал инъекцию зависимостей
- Поправил тесты связанные с изменениями в классе Lion
- В классе Feline поправил тесты, которые не используют аннотацию Spy
